### ISSUE TITLE
AC3.3: baro glitches can cause loss of attitude estimate

### ISSUE THREAD

**Issue by rmackay9** [diydrones/ardupilot#2619]
The vehicle can lose it's attitude estimate after an extreme baro glitch.  We've seen one case that appears (to me at least) to show this on the AC3.3 beta testing thread.

The bad attitude estimate can be reproduced in SITL by setting the SIM_BARO_ALT to -2000 waiting for 15 seconds, restore it to "0" wait another 15 seconds, then repeat.
https://dl.dropboxusercontent.com/u/3852647/AC3.3-baroglitch-sitltest/AC3.3-rc8-baro-glitch-test.bin

Tridge suggests that it could be the Accel Z bias estimate going bad.
![baroaltvsazbias](https://cloud.githubusercontent.com/assets/1498098/8950412/412b9f94-35fc-11e5-9aed-ce42be974954.png)

**Comment by rmackay9** [comment]
I've re-tested this and was unable to recreate attitude estimate problems after the EKF altitude was reset to the baro altitude.  What I saw instead was that if two very long (>15sec) glitches to the baro occured, the EKF altitude's accel-Z bias would go bad and the EKF altitude would not track the baro well.  After drifting too far from the baro, the EKF would reset it's altitude to the baro but it could lose 20m during this period.
In short, the current EKF can't handle two very long baro glitches in a row.  We're not going to try and fix this, maybe we will revisit this as part of the EKF rewrite (aka "New Maths").
